test/e2e/user_group_limit/user_group_limit_test.go

```diff
@@ -920,6 +1118,19 @@ var _ = ginkgo.Describe("UserGroupLimit", func() {
 	})
 })
 
+func GetUserInfoFromPodAnnotation(pod *v1.Pod) (interface{}, error) {
+	userInfo, ok := pod.Annotations[amCommon.UserInfoAnnotation]
+	if !ok {
+		return nil, fmt.Errorf("user info not found in pod annotation")
+	}
+	var userInfoObj interface{}
```

Comment:
We know the type of this object. It's `si.UserGroupInformation`, let's use that.

Reply:
Updated the code with requested changes.